app/cdap/components/GridTable/index.tsx

```diff
@@ -231,88 +312,231 @@ export default function GridTable() {
+              : from === 'undo' || arr?.length === 0
+              ? 'Transformation successfully deleted'
+              : `${removed_arr?.length} transformation successfully deleted from ${
+                  arr[arr.length - 1]
+                }`,
+        });
+      },
+      (err) => {
+        setSnackbar({
+          open: true,
+          isSuccess: false,
+          message: `Failed to transform ${newDirective}`,
+        });
+        setLoading(false);
+        setShowRecipePanel(false);
+      }
+    );
+  };
+
+  const deleteRecipes = (new_arr, remaining_arr) => {
+    applyDirectiveAPICall(new_arr, 'delete', remaining_arr, 'panel');
+    DataPrepStore.dispatch({
+      type: DataPrepActions.setUndoDirective,
+      payload: {
+        undoDirectives: [],
+      },
+    });
+  };
+
   useEffect(() => {
     getGridTableData();
   }, [gridData]);
 
+  const handleCloseSnackbar = () => {
+    setSnackbar(() => ({
+      open: false,
+    }));
+  };
+
   return (
     <Box data-testid="grid-table-container">
       <BreadCrumb datasetName={wid} />
-      {Array.isArray(gridData?.headers) && gridData?.headers.length === 0 ? (
-        <NoRecordScreen
-          title={T.translate('features.WranglerNewUI.NoRecordScreen.gridTable.title')}
-          subtitle={T.translate('features.WranglerNewUI.NoRecordScreen.gridTable.subtitle')}
-        />
-      ) : (
-        <Table aria-label="simple table" className="test">
-          <TableHead>
-            <TableRow>
-              {headersNamesList?.length &&
-                headersNamesList.map((eachHeader) => (
-                  <GridHeaderCell
-                    label={eachHeader.label}
-                    types={eachHeader.type}
-                    key={eachHeader.name}
-                  />
-                ))}
-            </TableRow>
-            <TableRow>
-              {missingDataList?.length &&
-                headersNamesList.length &&
-                headersNamesList.map((each, index) => {
-                  return missingDataList.map((item, itemIndex) => {
-                    if (item.name === each.name) {
-                      return <GridKPICell metricData={item} key={item.name} />;
-                    }
-                  });
-                })}
-            </TableRow>
-          </TableHead>
-          <TableBody>
-            {rowsDataList?.length &&
-              rowsDataList.map((eachRow, rowIndex) => {
-                return (
-                  <TableRow key={`row-${rowIndex}`}>
-                    {headersNamesList.map((eachKey, eachIndex) => {
-                      return (
-                        <GridTextCell
-                          cellValue={eachRow[eachKey.name] || '--'}
-                          key={`${eachKey.name}-${eachIndex}`}
-                        />
-                      );
+      <TablePanelContainer>
+        {Array.isArray(gridData?.headers) && gridData?.headers.length > 0 ? (
```

Comment:
this ternary operator returns huge blocks of code - please don't do that

Reply:
Done.